src/Raven.Server/Documents/Handlers/CollectionsHandler.cs

```diff
@@ -16,29 +15,56 @@ public class CollectionsHandler : DatabaseRequestHandler
         public Task GetCollectionStats()
         {
             using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
-            using (context.OpenReadTransaction())
             {
-                var collections = new DynamicJsonValue();
-                var result = new DynamicJsonValue
-                {
-                    [nameof(CollectionStatistics.CountOfDocuments)] = Database.DocumentsStorage.GetNumberOfDocuments(context),
-                    [nameof(CollectionStatistics.CountOfConflicts)] = 
-                        Database.DocumentsStorage.ConflictsStorage.GetNumberOfDocumentsConflicts(context),
-                    [nameof(CollectionStatistics.Collections)] = collections
-                };
+                CollectionStatistics collectionStatistics = new CollectionStatistics();
 
-                foreach (var collectionStat in Database.DocumentsStorage.GetCollections(context))
+                FillCollectionStats(collectionStatistics, context);
+
+                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                    context.Write(writer, collectionStatistics.ToJson());
+            }
+
+            return Task.CompletedTask;
+        }
+
+        [RavenAction("/databases/*/collections/stats/detailed", "GET", AuthorizationStatus.ValidUser)]
+        public Task GetDetailedCollectionStats()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                DetailedCollectionStatistics detailedCollectionStatistics = new DetailedCollectionStatistics();
+
+                FillCollectionStats(detailedCollectionStatistics, context);
+
+                using (context.OpenReadTransaction())
```

Comment:
Switch these lines and make `FillCollectionStats` expect an open tx, would save two tx here.